auth.php

```diff
@@ -542,6 +550,40 @@ public function saml_login() {
         return;
     }
 
+    /**
+     * Checks if the flagged user feature is enabled and if so, handles configured flag attribute
+     *
+     * @param $attributes
+     *
+     * @throws \moodle_exception
+     */
+
+    public function handle_flagged_login($attributes) {
+        global $redirect;
+
+        // Check if flagged login feature is enabled
+        if ($this->config->flagresponsetype != saml2_settings::OPTION_FLAGGED_LOGIN_NONE) {
+
+            $isflagactive = ($attributes[$this->config->flagattribute][0] == $this->config->flagvalue);
```

Comment:
I'll rework this to include more validation of the IdP attributes and to prevent empty string for the `flagattribute` or `flagvalue`. I may include logging so that it is easier to debug if an empty string is causing failure.